discord_bot/chat_handler.py

```
@@ -454,15 +456,31 @@ async def run_backfill_and_sync():
             except Exception as e:
                 logger.error(f"[on_ready] Backfill/sync task failed: {e}", exc_info=True)
 
+        nonlocal backfill_task
+        if backfill_task and not backfill_task.done():
+            logger.info("[on_ready] Backfill+sync task is already running; skipping duplicate startup")
+            return
+
         logger.info(
             f"[on_ready] Creating backfill+sync background task for {len(text_channels)} channels..."
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)